### ISSUE TITLE
After using HandleBar, Chinese comments are encoded

### ISSUE THREAD

**Issue by Varorbc** [ErikEJ/EFCorePowerTools#1093]
![image](https://user-images.githubusercontent.com/5714438/132320215-7eca2e29-922b-4476-b3e5-a1d8e9f702d3.png)

### Further technical details
EF Core Power Tools version: 2.5.750

Database engine: SQL Server

Visual Studio version: Visual Studio 2019 16.9

**Comment by Varorbc** [comment]
@ErikEJ I used the EntityFrameworkCore.Scaffolding.Handlebars test and found it working properly

**Comment by ErikEJ** [comment]
Could be due to this?

https://github.com/ErikEJ/EFCorePowerTools/blob/master/src/GUI/RevEng.Core/CommentCSharpEntityTypeGenerator.cs#L207

**Comment by ErikEJ** [comment]
@ajcvickers FYI! Potential issue with this new EF6 feature 😢

**Comment by ErikEJ** [comment]
@Varorbc What is you do not use Handlebars?

**Comment by Varorbc** [comment]
> Could be due to this?
> 
> https://github.com/ErikEJ/EFCorePowerTools/blob/ee99a4d80bc944a78ba648a67c50abd74422bfca/src/GUI/RevEng.Core/CommentCSharpEntityTypeGenerator.cs#L207
> 

Not here

**Comment by Varorbc** [comment]
> @Varorbc What is you do not use Handlebars?

EntityFrameworkCore.Scaffolding.Handlebars

**Comment by ErikEJ** [comment]
I meant - does the problem occur if you do not use Handlebars? (With EF Core Power Tools) ?

**Comment by Varorbc** [comment]
> I meant - does the problem occur if you do not use Handlebars? (With EF Core Power Tools) ?

not occur

**Comment by ErikEJ** [comment]
Ok, let's await a comment from @tonysneed

**Comment by Varorbc** [comment]
only use Handlebars work well

**Comment by ajcvickers** [comment]
@ErikEJ Can you elaborate on the connection to the EF Core code?

**Comment by ErikEJ** [comment]
@ajcvickers this is the implementation of database comments as XML comments on EF Core 6, but it turned out that everything works fine, if Handlebars are not part of the equation - in other words, sorry about false alarm

**Comment by tonysneed** [comment]
There is a bug in the version of Handlebars.net used by my scaffolding library where foreign characters are not escaped.

I can fix by using EF Core for comments when I upgrade by library to use .NET 6.

**Comment by Varorbc** [comment]
@tonysneed Why are some characters encoded, and some characters are not encoded?
![image](https://user-images.githubusercontent.com/5714438/132436086-8ea9c030-16f6-40bd-acfb-3fc501362fb3.png)

**Comment by tonysneed** [comment]
@Varorbc Behavior depends on the version of Handlebars.Net that is used. I re-opened the [issue](https://github.com/TrackableEntities/EntityFrameworkCore.Scaffolding.Handlebars/issues/30) on my repo, so that I will upgrade the Handlebars.Net package version to 2.0 or greater.

In the meantime, can you tell me what version of Handlebars.Net you are using?

**Comment by tonysneed** [comment]
This issue has been fixed by [168](https://github.com/TrackableEntities/EntityFrameworkCore.Scaffolding.Handlebars/pull/168) in my repo and will be released with EntityFrameworkCore.Scaffolding.Handlebars v5.0.3.

**Comment by ErikEJ** [comment]
Should be fixed in latest daily

**Comment by ErikEJ** [comment]
Had to revert the fix for now

**Comment by tonysneed** [comment]
Fixed #1109 with Handlebars release [v5.0.4](https://github.com/TrackableEntities/EntityFrameworkCore.Scaffolding.Handlebars/releases/tag/v5.04). You should be able to un-revert.

**Comment by ErikEJ** [comment]
Thanks a lot, @tonysneed !

**Comment by ErikEJ** [comment]
@Varorbc Fixed (again) in the latest daily, would be grateful if you could confirm

**Comment by Varorbc** [comment]
@ErikEJ reproduce the issue

**Comment by ErikEJ** [comment]
>> reproduce the issue

Sorry, I do not understand. Do you mean the issue is still present in the latest daily build? (Build 776 or later)

**Comment by Varorbc** [comment]
yes

![image](https://user-images.githubusercontent.com/5714438/134626342-23078100-17da-46f5-b254-23ddd415a506.png)

**Comment by ErikEJ** [comment]
Please file a new issue in the Handlebars repo, it isnot something I can help you with.

**Comment by Varorbc** [comment]
If this option is not checked, it works well

![Uploading image.png…]()

**Comment by tonysneed** [comment]
I have re-opened [issue 30](https://github.com/TrackableEntities/EntityFrameworkCore.Scaffolding.Handlebars/issues/30) on my repo. @Varorbc Could you please comment on it there to describe the behavior? Of course, a PR with a fix is welcome! 😄

**Comment by ErikEJ** [comment]
@tonysneed  Thanks for your continued support!

**Comment by tonysneed** [comment]
Published Handlebars [v5.0.5-preview1](https://github.com/TrackableEntities/EntityFrameworkCore.Scaffolding.Handlebars/releases/tag/v5.0.5-preview1), so that @ErikEJ can include it in a daily build and @Varorbc can test it.

**Comment by ErikEJ** [comment]
Included in latest daily build

**Comment by Varorbc** [comment]
still reproduce

**Comment by tonysneed** [comment]
@Varorbc Could you provide a repo to reproduce?

**Comment by Varorbc** [comment]
Ok, I will create a pr in your repo later

**Comment by Varorbc** [comment]
https://github.com/TrackableEntities/EntityFrameworkCore.Scaffolding.Handlebars/pull/179

**Comment by tonysneed** [comment]
@Varorbc See [PR 180](https://github.com/TrackableEntities/EntityFrameworkCore.Scaffolding.Handlebars/pull/180) in my repo. I set `HandlebarsLib.Configuration.NoEscape = true` in the template services. Comments on `Product` table and `ProductId` properties are encoded, but the comment on the `ProductName` property is not encoded. Even more strange, this only happens when data annotations are used.

**Comment by Varorbc** [comment]
Yes, this is what I have been unable to understand. Maybe we missed something

**Comment by tonysneed** [comment]
The behavior is inconsistent. Comments on the class name are encoded even with attributes. The `ProductId` property has a `[Key]` attribute and the comment is not encoded. But the next property, 'ProductName' has some attributes and is encoded. If there is no `#each property-annotations` in the template, then the all the property comments are encoded.

```csharp
namespace FakeNamespace
{
    /// <summary>
    /// 产品
    /// </summary>
    [Table("Product")]
    [Index(nameof(CategoryId), Name = "IX_Product_CategoryId")]
    public partial class Product
    {

        /// <summary>
        /// 编号
        /// </summary>
        [Key]
        public int ProductId { get; set; }

        /// <summary>
        /// &#21517;&#31216;
        /// </summary>
        [Required]
        [StringLength(40)]
        public string ProductName { get; set; }
        [Column(TypeName = "money")]
        public decimal? UnitPrice { get; set; }
        public bool Discontinued { get; set; }
        public byte[] RowVersion { get; set; }
        public int? CategoryId { get; set; }

        [ForeignKey(nameof(CategoryId))]
        [InverseProperty("Products")]
        public virtual Category Category { get; set; }
    }
}
```

To me this looks like a bug in [Handlebars.Net](https://github.com/Handlebars-Net/Handlebars.Net). I filed this [issue](https://github.com/Handlebars-Net/Handlebars.Net/issues/468) there.

**Comment by ErikEJ** [comment]
Closing as external, will update when fixed

**Comment by ErikEJ** [comment]
Fixed in latest daily
